server/test/controllers/ControllerTest.scala

```diff
@@ -58,27 +74,74 @@ class ControllerTest extends PlaySpec with MockFactory {
-        val dao = new Dao {
-          def addTask(task: NewTask) = Future.successful(())
-
-          def create(user: User) = ???
-
-          def getTasks(`type`: TaskType, limit: Port, yearAgo: Port) = ???
-
-          def getTask(year: Date, taskType: TaskType, timeuuid: UUID) = ???
-        }
+        val badSuite = "import org.scalatest.{FlatSpec, Matchers}\n\nclass SubArrayWithMaxSumTest(solution: SubArrayWithMaxSumSolution) extends FlatSpec with Matchers {\n  behavior of \"SubArrayWithMaxSum\"\n\n  it should \"return max sum sub array within given array\" in {\n    solution.apply(Array(-2, 1, -3, 4, -1, 2, 1, -5, 4)) should be(Array(4, -1, 2, 1))\n    solution.apply(Array(-2, 1, -3, 4, -1, 2, 1, 5, 4)) should be(Array(4, -1, 2, 1, 5, 4))\n    solution.apply(Array(2, -1, 0, 0, 0, 0, 1)) should be(Array(2))\n  }\n\n  it should \"return the whole array when given array has only positive numbers\" in {\n    solution.apply(Array(2, 1, 3, 4, 1, 2, 1, 5, 4)) should be(Array(2, 1, 3, 4, 1, 2, 1, 5, 4))\n  }\n\n  it should \"return max sum sub array when given array contains only negative numbers\" in {\n    solution.apply(Array(-2, -1, -3, -4, -1, -2, -1, -5, -4)) should be(Array(-1))\n    solution.apply(Array(-2, -3, -3, -4, -6, -2, -6, -5, -1)) should be(Array(-2))\n  }\n}\n\ntrait SubArrayWithMaxSumSolution {\n  def apply(a: Array[Int]): Array[Int]\n}"
+        val dao = stub[Dao]
+        //when
         withAddTaskController(new ScalaTestRunner, dao)({ controller =>
-          val result = controller.postNewTask(FakeRequest("POST", "ignore").withFormUrlEncodedBody("taskDescription" -> "1", "solutionTemplate" -> "2", "referenceSolution" -> solution, "suite" -> suite))
+          val result = controller.postNewTask(FakeRequest("POST", "ignore")
+            .withFormUrlEncodedBody("taskName" -> "0", "taskDescription" -> "1",
+              "solutionTemplate" -> "2", "referenceSolution" -> solution, "suite" -> badSuite))
+          //then
           status(result) mustBe BAD_REQUEST
           contentAsString(result) must include( """class="error"></dd>""")
+          dao.addTask _ verify NewTask(scalaClass, "0", "1", "2", solution, badSuite) never()
```

Comment:
FYI: contentAsString(result)  does not contain error message from messageApi. Looks like on Play limitation.